From ec2c4ead497133ef20d5ef5a9b481b38e1e0f7a2 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Mon, 27 Jun 2011 13:13:54 +1000
Subject: Assign local->private after allocating.

It is detrimental to the user experience when the driver tries to derefernce
null pointers.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>

diff --git a/src/acecad.c b/src/acecad.c
index ad6d793..604fe91 100644
--- src/acecad.c
+++ src/acecad.c
@@ -343,6 +343,7 @@ AceCadPreInit(InputDriverPtr drv, InputInfoPtr local, int flags)
         return BadAlloc;
 
     memset(priv, 0, sizeof(AceCadPrivateRec));
+    local->private = priv;
 
     local->device_control = DeviceControl;
 
-- 
cgit v0.10.2

